### PR TITLE
Tech CORE-422 | Refactor of the error structure in the project

### PR DESCRIPTION
**Pull request by nicholaspcr**
# Description

Kind: Tech

Section: pkg/ierrors

Summary: refactored the ierrors pkg and all the related pkgs that used their functions

### CODE REVIEW

**Comment by luiz-couto** [comment]
Maybe a documentation of how to use pproof would be nice too

File: pkg/ierrors/doc.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// - Wrap/Unwrap functions: handles the addition and removal of context to an
+// error, meaning that if desired the developer can provide extra information to
+// the error he can simply use `ierrors.Wrap(err, "on operation X")`.
+//
+// - Marshal/Unmarshal: the ierror structure declared by this package have
+// custom functions that are called when using json.Marshal or json.Unmarshal,
+// meaning that when creating a error it can be parsed into json and yaml
+// format, allowing the transfer of error information via http.
+//
+// - From: it receives an external error interace and converts it to an ierror
```
Comment (luiz-couto) [inline pkg/ierrors/doc.go]: "interface" misspelling

**Review by LucasStarlingdePaulaSalles — APPROVED** [review]

File: cmd/insprd/memory/fake/alias.go [hunk 2]
```diff
@@ -56,11 +53,7 @@ func (a *Alias) Delete(scope, aliasKey string) error {
 
 	_, ok := a.alias[scope]
 	if !ok {
-		return ierrors.
-			NewError().
-			NotFound().
-			Message("type %s not found", scope).
-			Build()
+		return ierrors.New(fmt.Sprintf("type %s not found", scope)).NotFound()
```
Comment (lucasABLima) [inline cmd/insprd/memory/fake/alias.go]: Is it really necessary to use this fmt.Sprintf?

Comment (nicholaspcr) [reply cmd/insprd/memory/fake/alias.go]: Not anymore, i agree that it was inappropriate so i decided to add the possibility of just passing string as a format and its values.

`ierrors.New("%v-%v', var1, var2)` is working and is tested

**Review by luiz-couto — APPROVED** [review]

**Review by lucasABLima — APPROVED** [review]

**Review by ptcar2009 — APPROVED** [review]
lgtm

**Review by LucasStarlingdePaulaSalles — APPROVED** [review]

**Pull request merged**